client/etcd/client.go

```diff
@@ -240,8 +243,31 @@ func (c *csclient) etcdClientGen(zone string) (*clientv3.Client, error) {
 	return cli, nil
 }
 
-func newClient(endpoints []string) (*clientv3.Client, error) {
-	return clientv3.New(clientv3.Config{Endpoints: endpoints})
+func newClient(endpoints []string, certfile string, keyfile string, ca string) (*clientv3.Client, error) {
+	tlscfg := &tls.Config{}
+	if certfile != "" {
```

Comment:
suggestion(take/leave): group the cert specific code into a function on the sub-type for the config. 

i.e. 
```
type TLSConfiguration struct {
  KeyPath string `yaml...`
  CrtPath ... 
  CACrtPath .. 
} 

func (t TLSConfiguration) NewTLSConfig() (*tls.Config, error) {
  ... 
}
```

Reply:
I'd agree with you if there were any other logic in the newClient function beforehand, but as is there's nothing, so all we'd do is add another layer of if err != nil boilerplate around the whole thing. Would prefer to leave as-is.

Reply:
I'd +1 to Prateek's point, the tls logic is pretty isolated, if we ever need to change the tls logic we just need to add fields to the TLSConfig struct without having to add another field to this function.

Reply:
Like I said, there's no logic to separate- before the TLS logic this function was just a return statement. There's nothing else having to do with TLS that etcd cares about- almost all the other fields available in the tls.Config object are constrained by GRPC's restrictions on TLS, and we'd have to make the get/set fields in the tls config object anyhow, unless I'm misunderstanding your suggestion.